### PR TITLE
[Import subscriber] Added support to merge exisitng subscribers' attributes

### PR DESCRIPTION
**Pull request by srishti-vishnoi**
**Issue:**
Currently, when importing subscribers via CSV, the system only allows for overwriting or skipping existing subscribers. The system cannot merge user attributes with existing ones.

**Proposed Solution:**
To address this limitation, I've implemented support for merging user attributes during CSV import. Users now have the option to choose whether to merge attributes when overwriting is set to False.

**Key Points:**

Overwrite? | Merge? | Expected Result
-- | -- | --
TRUE | TRUE | Merge Ignored  (toggle hidden) (NOT POSSIBLE)
TRUE | FALSE | Merge Ignored (toggle hidden) (NOT POSSIBLE)
FALSE | TRUE | Subscribers' data will be merged
FALSE | FALSE | Subscribers will be skipped during import (No change)

*Additional Notes:*

1. When overwrite is set to True, the Merge toggle will be hidden and the value of the Merge flag will be set to False.
2. The Import subscriber API will now throw a Bad Request error if both merge and overwrite are set to True, preventing conflicting settings.

This enhancement ensures a more flexible and intuitive subscriber management process during CSV imports.

Screenshots:
Screenshot from 2024-02-26 16-38-41

### CODE REVIEW

File: cmd/import.go [hunk 1]
```diff
@@ -6,7 +6,6 @@ import (
 	"os"
 	"net/http"
 	"strings"
-
```
Comment (sandipagarwal) [inline cmd/import.go]: Don't make a change that is not required.

File: i18n/en.json [hunk 2]
```diff
@@ -256,6 +257,8 @@
     "import.mode": "Mode",
     "import.overwrite": "Overwrite?",
     "import.overwriteHelp": "Overwrite name, attribs, subscription status of existing subscribers?",
+    "import.merge": "Merge?",
+    "import.mergeHelp": "Merge attribs of existing subscribers?",
```
Comment (sandipagarwal) [inline i18n/en.json]: `Merge attributes of existing subscribers?`

File: queries.sql [hunk 3]
```diff
@@ -111,7 +111,11 @@ WITH sub AS (
     ON CONFLICT (email)
     DO UPDATE SET
         name=(CASE WHEN $7 THEN $3 ELSE s.name END),
-        attribs=(CASE WHEN $7 THEN $4 ELSE s.attribs END),
+        attribs=(CASE
+            WHEN $7 THEN $4
+            WHEN $8 THEN s.attribs || $4
+            ELSE s.attribs END
+        ),
```
Comment (sandipagarwal) [inline queries.sql:122]: Please update the comments that are present above this query.

File: queries.sql [hunk 4]
```diff
@@ -111,7 +111,11 @@ WITH sub AS (
     ON CONFLICT (email)
     DO UPDATE SET
         name=(CASE WHEN $7 THEN $3 ELSE s.name END),
-        attribs=(CASE WHEN $7 THEN $4 ELSE s.attribs END),
+        attribs=(CASE
+            WHEN $7 THEN $4
+            WHEN $8 THEN s.attribs || $4
+            ELSE s.attribs END
```
Comment (sandipagarwal) [inline queries.sql:121]: Can this be optimised?

**Review by sandipagarwal — CHANGES_REQUESTED** [review]

File: queries.sql (hunk 4 above)
Comment (sandipagarwal) [reply queries.sql:121]: Not required anymore.

Comment (srishti-vishnoi) [reply queries.sql:121]: As discussed, this is no more required

**Review by sandipagarwal — APPROVED** [review]

**Pull request merged**